src/Auth.ts

```diff
@@ -286,40 +291,78 @@ export class Auth {
 
       let cert: string = '';
 
-      if (this.service.password === undefined) {
-        const buf = Buffer.from(this.service.certificate as string, 'base64');
-        cert = buf.toString('utf8');
-      }
-      else {
-        const buf = Buffer.from(this.service.certificate as string, 'base64');
-        const p12Asn1 = asn1.fromDer(buf.toString('binary'), false);
-
-        const p12Parsed = pkcs12.pkcs12FromAsn1(p12Asn1, false, this.service.password);
-
-        let keyBags: any = p12Parsed.getBags({ bagType: pki.oids.pkcs8ShroudedKeyBag });
-        const pkcs8ShroudedKeyBag = keyBags[pki.oids.pkcs8ShroudedKeyBag][0];
+      if (this.service.thumbprint && this.service.certificate === undefined && this.service.password === undefined ) {
```

Comment:
The idea of service is to persist the information about the certificate, so ideally, we wouldn't persist here the fact that we retrieve the cert from a windows cert store but rather the retrieved cert so that we don't need to reach out to the cert store on each command execution which would hurt performance.

Instead, we should move the logic of retrieving the cert to the `login` command and passed the retrieved cert onwards following the same approach as we already have implemented when using a certificate file: https://github.com/pnp/cli-microsoft365/blob/7d5d60e556b1cde4854a71876156831915032b45/src/m365/commands/login.ts#L71